lib/experimental/fonts/font-library/font-library.php

```diff
@@ -33,5 +33,20 @@ function gutenberg_init_font_library() {
 	$font_library_controller->register_routes();
 }
 
-add_action( 'rest_api_init', 'gutenberg_init_font_library' );
+add_action( 'rest_api_init', 'gutenberg_init_font_library_routes' );
 
+/**
+ * Registers the font collection filter for font library.
+ *
+ * @since 6.4.0
+ */
+function gutenberg_add_register_font_collection_filter() {
+	add_filter(
+		'wp_register_font_collection',
+		array( 'WP_Font_Library', 'register_font_collection' ),
+		10,
+		2
+	);
+}
+
+add_action( 'init', 'gutenberg_add_register_font_collection_filter' );
```

Comment:
Why does it need to add the filter on `'init'`? Why not move the `add_filter` hook out of the function like this?

```suggestion
add_filter( 'wp_register_font_collection', array( 'WP_Font_Library', 'register_font_collection' ), 10, 2 );
```

Reply:
Good suggestion!